bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -312,12 +407,13 @@ static ICoreWebView2CookieManager getCookieManager() {
 		SWT.error(SWT.ERROR_NOT_IMPLEMENTED, null, " [WebView2: cookie access requires a Browser instance]");
 	}
 	Edge instance = environmentWrapper.instances().get(0);
-	if (instance.webView_2 == null) {
+	instance.webViewProvider.waitForPendingWebviewTasksToFinish();
```

Comment:
Now all except this calls to `waitForPendingWebiewTaskToFinish()` are encapsulated within the `WebViewProvider`. I propose to keep that functionality completely hidden in the `WebViewProvider`. I guess calls to `isWebView*Unavailable()` should probably block until initialization has finished, as otherwise the method will return unreasonable resonable in case the web view has not been initialized yet anyway.

Reply:
cant put waitForPendingWebiewTaskToFinish inside isWebView*Unavailable cause its used in callbacks and calling it inside that would cause a deadlock

Reply:
But actually that's not even what you want to do here, is it? You want to wait for webView being initialized, not for all tasks being finished. That is something that should be done in the `isWebView*Unavailable()` methods anyway in order to avoid that they return unreasonable results when called while the web view has not been initialized yet.

In any case, hiding the need to wait for pending tasks in all but one case is a design flaw for which we need to have a solution.

Reply:
I moved it inside the method and tested it. It seems fine and it isnt causing any deadlock. Seems good.